tls/s2n_kem.c

```diff
@@ -23,7 +23,18 @@
 #include "utils/s2n_mem.h"
 #include "utils/s2n_safety.h"
 
-/* The KEM IDs and names come from https://tools.ietf.org/html/draft-campagna-tls-bike-sike-hybrid */
+const struct s2n_kem s2n_mlkem_768 = {
+    .name = "mlkem768",
+    .kem_nid = S2N_NID_MLKEM768,
+    .kem_extension_id = 0, /* This is not used in TLS 1.2's KEM extension */
+    .public_key_length = S2N_MLKEM_768_PUBLIC_KEY_BYTES,
+    .private_key_length = S2N_MLKEM_768_SECRET_KEY_BYTES,
+    .shared_secret_key_length = S2N_MLKEM_768_SHARED_SECRET_BYTES,
+    .ciphertext_length = S2N_MLKEM_768_CIPHERTEXT_BYTES,
+    .generate_keypair = &s2n_evp_kem_generate_keypair,
+    .encapsulate = &s2n_evp_kem_encapsulate,
+    .decapsulate = &s2n_evp_kem_decapsulate,
```

Comment:
Not a blocker for this PR, but maybe a follow-up issue: if all the s2n_kems use the same generate/encapsulate/decapsulate functions, do they need to each specify those function pointers? Or should we just assume the function and simplify these definitions?

Reply:
These were originally used to select between the interned Kyber vs the LibCrypto Kyber, so right now they are all the same. However, if we want s2n-tls to have ML-KEM support with BoringSSL we will probably need to override these function pointers to BoringSSL specific ones as BoringSSL doesn't expose ML-KEM over it's EVP API's but instead offers "raw" MLKEM KeyGen/Encap/Decap API's. 
 - https://github.com/google/boringssl/blob/master/include/openssl/mlkem.h